src/server/lib/jobs.ts

```diff
@@ -0,0 +1,25 @@
+import { addToGroup, GroupCode, updateName } from './idapi/user';
+
+const setupJobsUser = async (
+  firstName: string,
+  secondName: string,
+  ip: string,
+  sc_gu_u: string,
+) => {
+  if (firstName === '' || secondName === '') {
+    throw new Error('Empty values not permitted for first or last name.');
+  }
+  // When a jobs user is registering, we add them to the GRS group and set their full name.
+  // We also want to set their first and last name—these are required fields for all Jobs users.
+  //
+  // Once users belong to the GRS group, they aren't shown the /accept/GRS page when
+  // they try to sign in to the Jobs site for the first time.
+  //
+  // We can resolve both promises here because they are not dependent on each other.
+  return await Promise.all([
+    updateName(firstName, secondName, ip, sc_gu_u),
+    addToGroup(GroupCode.GRS, ip, sc_gu_u),
+  ]);
+};
```

Comment:
you don't actually need the `async/await` here, as where this method is already returning a promise, so is being used is already calling `await` at that level, there's other places in the project that has used this antipattern too, e.g. the okta api calls, so it's worth starting to address this at least here anyway

```suggestion
const setupJobsUser = (
  firstName: string,
  secondName: string,
  ip: string,
  sc_gu_u: string,
) => {
  if (firstName === '' || secondName === '') {
    throw new Error('Empty values not permitted for first or last name.');
  }
  // When a jobs user is registering, we add them to the GRS group and set their full name.
  // We also want to set their first and last name—these are required fields for all Jobs users.
  //
  // Once users belong to the GRS group, they aren't shown the /accept/GRS page when
  // they try to sign in to the Jobs site for the first time.
  //
  // We can resolve both promises here because they are not dependent on each other.
  return Promise.all([
    updateName(firstName, secondName, ip, sc_gu_u),
    addToGroup(GroupCode.GRS, ip, sc_gu_u),
  ]);
};
```

Reply:
ah yes, good spot, much better this way